app/views/network_router/_common_new_edit.haml

```diff
@@ -1,76 +1,121 @@
+                            :name          => "name",
+                            "ng-model"     => "vm.networkRouterModel.name",
+                            "ng-maxlength" => 128,
+                            "required"     => ""}
+        %span.help-block{"ng-show" => "angularForm.name.$error.required"}
+          = _("Required")
+  %h3
+    = _('External Gateway')
+  .form-horizontal
     .form-group
+      %label.col-md-2.control-label{"for" => "network_router_enable_snat"}
+      .col-md-8
+        %input{"bs-switch"   => "",
+               :data         => {:on_text => _('Yes'), :off_text => _('No'), :size => 'mini'},
+               "type"        => "checkbox",
+               "id"          => "network_router_external_gateway",
+               "name"        => "external_gateway",
+               "ng-model"    => "vm.networkRouterModel.external_gateway"}
+
+  .form-horizontal{"ng-if" => "vm.networkRouterModel.external_gateway"}
+    .form-group{"ng-if" => "vm.networkRouterModel.ems_id"}
+      %label.col-md-2.control-label{"for" => "network_router_enable_snat"}
+        = _('Enable Source NAT')
+      .col-md-8
+        %input{"bs-switch"   => "",
+               :data         => {:on_text => _('Yes'), :off_text => _('No'), :size => 'mini'},
+               "type"        => "checkbox",
+               "id"          => "network_router_enable_snat",
+               "name"        => "enable_snat",
+               "ng-model"    => "vm.networkRouterModel.enable_snat"}
+
+    .form-group{"ng-class" => "{'has-error': angularForm.cloud_network_id.$invalid}"}
       %label.col-md-2.control-label
-        = _('Subnet')
+        = _('Network')
       .col-md-8
-        %select{"name"                        => "cloud_subnet_id",
-                "ng-model"                    => "networkRouterModel.cloud_subnet_id",
-                :checkchange                  => true,
-                "ng-options"                  => "subnet.id as subnet.name for subnet in available_subnets",
+        %select{"name"                        => "cloud_network_id",
+                "ng-model"                    => "vm.networkRouterModel.cloud_network_id",
+                "ng-options"                  => "network.id as network.name for network in vm.available_networks",
+                "ng-change"                   => "vm.filterCloudNetworkChanged(vm.networkRouterModel.cloud_network_id)",
                 "pf-select"                   => true,
-                "ng-selected"                 => "networkRouterModel.cloud_subnet_id",
+                "ng-selected"                 => "vm.networkRouterModel.cloud_network_id",
+                "ng-required"                 => "vm.networkRouterModel.enable_snat == true",
                 "selectpicker-for-select-tag" => ""}
           %option{"value" => ""}
             = "<#{_('Choose')}>"
+        %span.help-block{"ng-show" => "angularForm.cloud_network_id.$error.required"}
+          = _("Required")
+
+    %h2
```

Comment:
Empty h2, then a line, and then a text that should go in that h2 :).

But it was here before too, so, merging regardless..